### ISSUE TITLE
W1b: `Boxes` conversion silently corrupts sub-unit extents — normalized-coordinate boxes do not round-trip

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4061]
`Boxes` stores extents inclusively and materializes vertices with `_boxes_to_polygons`, which places the top-right corner at `xmin + width - 1` (`boxes.py:88-92`). That `- 1` is only meaningful for integer pixel extents of at least one unit. For `0 < extent < 1` the "top-right" vertex lands **left** of "top-left" and the "bottom-right" vertex lands **above** "top-right": the stored quadrilateral is inverted on both axes and is no longer in the documented clockwise TL/TR/BR/BL order. `to_tensor` then re-derives the box with `amin`/`amax`, so it recovers a different — and larger — box than the caller supplied.

The affected modes are exactly the three that convert: `'xyxy'`, `'xywh'` and `'vertices'`. The `'_plus'` modes are unaffected because their `+ 1` cancels the `- 1`.

This is not an exotic input. It is *every* box in normalized `[0, 1]` coordinates, and it reaches the public augmentation API: `AugmentationSequential(..., data_keys=["bbox_xywh"])` corrupts such boxes through an otherwise identity pipeline. `validate_boxes=True` does not catch it, because `0.5 > 0` passes the only check there is.

## Repro

```python
import torch
import kornia.augmentation as K
from kornia.geometry.boxes import Boxes

cases = {
    "xyxy":          torch.tensor([[0.1, 0.1, 0.6, 0.9]]),
    "xywh":          torch.tensor([[0.1, 0.1, 0.5, 0.8]]),
    "vertices":      torch.tensor([[[0.1, 0.1], [0.6, 0.1], [0.6, 0.9], [0.1, 0.9]]]),
    "vertices_plus": torch.tensor([[[0.1, 0.1], [0.6, 0.1], [0.6, 0.9], [0.1, 0.9]]]),
}
for mode, t in cases.items():
    out = Boxes.from_tensor(t, mode=mode, validate_boxes=True).to_tensor(mode)
    print(f"{mode:14s} {[round(v, 2) for v in t.flatten().tolist()]} -> "
          f"{[round(v, 2) for v in out.flatten().tolist()]}")

print("stored:", Boxes.from_tensor(cases["xyxy"], mode="xyxy").data.round(decimals=2).tolist())

# end to end, through an identity pipeline
img = torch.rand(1, 3, 8, 8)
for key, bbox in [("bbox_xywh", torch.tensor([[[0.1, 0.1, 0.5, 0.8]]])),
                  ("bbox_xyxy", torch.tensor([[[0.1, 0.1, 0.6, 0.9]]]))]:
    aug = K.AugmentationSequential(K.Resize((8, 8), p=1.0), data_keys=["input", key])
    _, out = aug(img, bbox)
    print(f"{key:10s} {[round(v, 2) for v in bbox.flatten().tolist()]} -> "
          f"{[round(v, 2) for v in out.flatten().tolist()]}")
```

## Observed

```text
xyxy           [0.1, 0.1, 0.6, 0.9] -> [-0.4, -0.1, 1.1, 1.1]
xywh           [0.1, 0.1, 0.5, 0.8] -> [-0.4, -0.1, 1.5, 1.2]
vertices       [0.1, 0.1, 0.6, 0.1, 0.6, 0.9, 0.1, 0.9] -> [-0.4, -0.1, 1.1, -0.1, 1.1, 1.1, -0.4, 1.1]
vertices_plus  [0.1, 0.1, 0.6, 0.1, 0.6, 0.9, 0.1, 0.9] -> [0.1, 0.1, 0.6, 0.1, 0.6, 0.9, 0.1, 0.9]

stored: [[[0.1, 0.1], [-0.4, 0.1], [-0.4, -0.1], [0.1, -0.1]]]     <- TR is left of TL, BR is above TR

bbox_xywh  [0.1, 0.1, 0.5, 0.8] -> [-0.4, -0.1, 1.5, 1.2]          <- identity pipeline
bbox_xyxy  [0.1, 0.1, 0.6, 0.9] -> [0.1, 0.1, 0.6, 0.9]
```

A `0.5 x 0.8` box inside the unit square comes back as a `1.9 x 1.3` box that extends outside it on all four sides. No error, no warning. Integer-extent boxes and extents above one unit are unaffected (`[1., 2., 5., 4.]` round-trips exactly), which is why every existing test misses this: the suite uses integer pixel coordinates throughout.

`AugmentationSequential` is only partly exposed — `_preproc_boxes` maps `BBOX` to `'vertices_plus'` and `BBOX_XYXY` to `'xyxy_plus'`, both of which cancel — but `BBOX_XYWH` maps to plain `'xywh'` and is corrupted.

## Expected

A conversion round-trip preserves the box, or refuses the input. Three shapes, in increasing order of cost:

1. **The #3934 repair subsumes this.** With exclusive arithmetic there is no `- 1`: `_boxes_to_polygons` would place TR at `xmin + width`, `to_tensor` would drop the `+ 1`, and `[0.1, 0.1, 0.6, 0.9]` round-trips by construction. This is the continuous-coordinate face of the same defect.
2. Tighten `validate_boxes` so the non-`_plus` `'xy*'` modes reject `0 < extent < 1` instead of silently inverting. Turns silent corruption into an error, but does not fix `mode='vertices'` (unvalidated) and still breaks currently-accepted calls.
3. Document the restriction and leave the arithmetic alone — which is what the current docs implicitly do, and what makes normalized-coordinate use a trap.

Option 1 is the coherent one, and it is already scheduled.

## Classification

**Semantic contract / silent value corruption.** The inclusive `- 1` offset encodes an assumption — extents are integer pixel counts of at least one — that no signature, docstring or guard states. Continuous-coordinate input violates it silently and the `amin`/`amax` recovery in `to_tensor` launders the inverted quadrilateral into a plausible-looking wrong answer.

## Disposition

**MERGE IN WINDOW — with #3934.** Canonical: exclusive extent arithmetic, at which point this issue closes without a separate repair. Divergence today (executed): `Boxes.from_tensor(x, mode='xyxy').to_tensor('xyxy') != x` for any extent below one unit; `AugmentationSequential(data_keys=['bbox_xywh'])` inherits it. Not to be fixed ahead of the coordinated Repair & Deprecation window — a PR that changes `_boxes_to_polygons` on its own will be closed and pointed at #3934, since the two must move together.

Batch 4a (#4060) should pin the current behavior and anchor this issue, rather than stating the unconditional round-trip guarantee it currently does: `Boxes`'s new convention block asserts "Axis-aligned rectangles round-trip exactly in their own mode", and `test_convention_axis_aligned_box_round_trips_in_each_mode` uses integer coordinates with extents `(4, 2)`, so the pin cannot fail.

## Environment

kornia 0.9.0rc1 @ `4ab79c78` (main), torch 2.9.1, python 3.11.14, cpu. Found while reviewing #4060 during the bbox/boxes conventions audit (batch 4, tracking issue #4020); not in the up-front audit list.

Posted on behalf of @ducha-aiki by Claude (Opus 5).
